_posts/2019-04-16-preservation.md

```diff
@@ -0,0 +1,377 @@
+---
+date: 2019-04-16 8:56:00 -0400
+title: Preservation
+layout: default
+---
+
+## Preservation
+**by Trey Pendragon**
+
+Lately we've been working on preservation within
+[Figgy](https://github.com/pulibrary/figgy) and have come up with a strategy we
+think works for us. However, in doing so, we've evaluated a few options and come
+up with some assumptions which could be useful to others.
+<!--more-->
+
+This isn't our entire preservation story (there are other issues such as
+replication, management, audit trails, etc), but it compares a few options for
+our back-end storage.
+
+
+### Object Boundaries
+
+I talk about an "object" a lot below. After much discussion, we decided at
+  Princeton that we want a "directory" to consist of a top-level object (like a
+  Book) and all its children.
```

Comment:
Outsider comment - feel free to ignore. It'd be interesting to see some of that discussion: the pros and cons of requiring children to be nested under the parent directory.

Reply:
@bcail the tl;dr version is:

pro:
1. Packaging an object and its children together makes intuitive sense.
1. We expect restoration to happen at the object level, so packaging them together will make it easier to do that.
1. In disaster scenarios where Figgy isn't available, it would be much easier to work with a filesystem hierarchy that at least contained all the files and metadata for an object.  In cases where we have gone into Figgy's storage, it's very cumbersome to have to navigate around based on ids.

con:
1. What exactly is the boundary between an object and other objects?  Membership relationships usually indicate that the related things should be included, but not always (and we have modeled collection-like relationships both as members and in other ways)
1. Where do we stop? Do we want to also ingest controlled vocabulary terms? The whole controlled vocabulary hierarchy? Collections? How much of the repository do we want to duplicate, and would that serve any purpose?
1. Combining things and nesting them will make sync more challenging — if we had a 1-to-1 mapping between Figgy objects and Preservation objects, it would be simpler. But then what is the point of packaging at all? Why not just mirror the local file and metadata storage in the cloud and call it a day?